File: spring-boot/Chat2Learn/src/main/java/com/project/chat2learn/service/model/request/RegisterRequest.java
package com.project.chat2learn.service.model.request;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RegisterRequest {

    private String name;

    private String lastname;

    private String email;

    private String password;

    private String confirmPassword;
}

File: chat2learnfe/lib/model/request/RegisterRequest.dart
class RegisterRequest {
  String? name;
  String? lastname;
  String? email;
  String? password;
  String? confirmPassword;

  RegisterRequest(
      {this.name,
      this.lastname,
      this.email,
      this.password,
      this.confirmPassword});

  RegisterRequest.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    lastname = json['lastname'];
    email = json['email'];
    password = json['password'];
    confirmPassword = json['confirmPassword'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['name'] = this.name;
    data['lastname'] = this.lastname;
    data['email'] = this.email;
    data['password'] = this.password;
    data['confirmPassword'] = this.confirmPassword;
    return data;
  }
}
